scripts/bootloader/zb_add_ota_header.py

```diff
@@ -0,0 +1,204 @@
+
+        self.binary = ota_header.header + firmware
+        self.filename = '-'.join(['{:04X}'.format(manufacturer_code),
+                                  '{:04X}'.format(image_type),
+                                  '{:08X}'.format(file_version),
+                                  comment]) + '.zigbee'
+
+
+class OTA_header:
+    def __init__(self,
+                 file_id,
+                 header_version,
+                 header_length,
+                 header_field_control,
+                 manufacturer_code,
+                 image_type,
+                 file_version,
+                 zigbee_stack_version,
+                 header_string,
+                 total_image_size,
+                 min_hw_version=None,
+                 max_hw_version=None):
+
+        self.__pack_format = '<LHHHHHLHc31sL'
+        self.__header_length = header_length
+        self.__total_size = total_image_size
+        self.__field_control = header_field_control
+        self.__additional_fields = []
+
+        # If min and max hardware version are present add optional fields to the header
+        if (isinstance(min_hw_version, int) and isinstance(max_hw_version, int)):
+            self.__add_optional_fields((OTA_HEADER_MIN_HW_VERSION_LENGTH + OTA_HEADER_MAX_HW_VERSION_LENGTH),
+                                       OTA_HEADER_FIELD_CONTROL_BIT_MASK_HW_VER,
+                                       'HH',
+                                       [min_hw_version, max_hw_version])
+
+        self.__pack_args = [file_id,
+                            header_version,
+                            self.__header_length,
+                            self.__field_control,
+                            manufacturer_code,
+                            image_type,
+                            file_version,
+                            zigbee_stack_version,
+                            bytes([len(header_string)]),
+                            bytes(header_string.encode('ascii')),
+                            self.__total_size]
+        self.__pack_args.extend(self.__additional_fields)
+
+    @property
+    def header(self):
+        return struct.pack(self.__pack_format, *self.__pack_args)
+
+    def __add_optional_fields(self, fields_length, field_control_bit_mask, fields_formatting, fields_values):
+        self.__total_size    += fields_length
+        self.__header_length += fields_length
+        self.__field_control += field_control_bit_mask
+        self.__pack_format   += fields_formatting
+        if type(fields_values) in (tuple, list):
+            self.__additional_fields.extend(fields_values)
+        else:
+            self.__additional_fields.append(fields_values)
+
+def convert_version_string_to_int(s):
+    """Convert from semver string "1.2.3", to integer 10203"""
+    numbers = s.split(".")
+    if len(numbers) != 3:
+        raise print("application-version-string parameter must be on the format x.y.z")
+    js = [10000, 100, 1]
```

Comment:
Since every part of symver can be up to 1024, I would recommend to change js to `js=[1024*1024, 1024, 0]`. 
File version in OTA header is uint32 type, so generated version should fit.
Btw. ZCL spec contains recommended file version definition. One octect for each:
* Application Release 
* Application Build 
* Stack Release 
* Stack Build